src/storage/user-data.ts

```diff
@@ -119,29 +119,43 @@ export async function saveUserData( data: UserData ): Promise< void > {
 function toDiskFormat( { sites, ...rest }: UserData ): PersistedUserData {
 	return {
 		version: 1,
-		sites: sites.map( ( { id, path, adminPassword, port, phpVersion, name, themeDetails } ) => {
-			// No object spreading allowed. TypeScript's structural typing is too permissive and
-			// will permit us to persist properties that aren't in the type definition.
-			// Add each property explicitly instead.
-			const persistedSiteDetails: PersistedUserData[ 'sites' ][ number ] = {
+		sites: sites.map(
+			( {
 				id,
-				name,
 				path,
 				adminPassword,
 				port,
 				phpVersion,
-				themeDetails: {
-					name: themeDetails?.name || '',
-					path: themeDetails?.path || '',
-					slug: themeDetails?.slug || '',
-					isBlockTheme: themeDetails?.isBlockTheme || false,
-					supportsWidgets: themeDetails?.supportsWidgets || false,
-					supportsMenus: themeDetails?.supportsMenus || false,
-				},
-			};
+				name,
+				themeDetails,
+				customDomain,
+				useCustomDomain,
+			} ) => {
+				// No object spreading allowed. TypeScript's structural typing is too permissive and
+				// will permit us to persist properties that aren't in the type definition.
+				// Add each property explicitly instead.
+				const persistedSiteDetails: PersistedUserData[ 'sites' ][ number ] = {
+					id,
+					name,
+					path,
+					adminPassword,
+					port,
+					phpVersion,
+					customDomain,
+					useCustomDomain,
```

Comment:
Do we really need to store a boolean for `useCustomDomain` in the user config? Is it not enough to store `customDomain`?

Reply:
I'm hesitant, I believe that we can get away with just "customDomain" but a boolean gives a sense of clarity. We absolutely need both as state values in the form.